hack/generator/pkg/functions/resource_conversion_function_test.go

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ */
+
+package functions
+
+import (
+	"testing"
+
+	. "github.com/onsi/gomega"
+
+	"github.com/Azure/azure-service-operator/hack/generator/pkg/astmodel"
+	"github.com/Azure/azure-service-operator/hack/generator/pkg/conversions"
+	"github.com/Azure/azure-service-operator/hack/generator/pkg/test"
+)
+
+// Test_ResourceConversionFunction_DirectConversion_GeneratesExpectedCode tests the code when the ConvertTo() and
+// ConvertFrom() functions are directly converting to/from the Hub type, without any intermediate step.
+func Test_ResourceConversionFunction_DirectConversion_GeneratesExpectedCode(t *testing.T) {
```

Comment:
Same question if we can use one of the standard json file based golden tests once this pipeline stage is actually fully enabled.

If so leave a TODO here as well?

Reply:
No. The purpose of the test is to verify the behavour of the stage in isolation, to help troubleshoot any issues as specific to this stage.